libmachine/provision/ubuntu.go

```diff
@@ -124,7 +124,7 @@ func (provisioner *UbuntuProvisioner) Hostname() (string, error) {
 
 func (provisioner *UbuntuProvisioner) SetHostname(hostname string) error {
 	cmd, err := provisioner.SSHCommand(fmt.Sprintf(
-		"sudo hostname %s && echo %q | sudo tee /etc/hostname && echo \"127.0.0.1 %s\" | sudo tee -a /etc/hosts",
+		"sudo hostname %s && echo %q | sudo tee /etc/hostname && sudo sed -i 's/^127.0.1.1.*/127.0.1.1 %s/g' /etc/hosts",
```

Comment:
@crunchywelch Why the additional `sed` command instead of just replacing 127.0.0.1 with 127.0.1.1?